### PR TITLE
gRPC conversions - Nodes

### PR DESCRIPTION
**Pull request by Joerger**
Convert Node endpoints to gRPC with ctx.

Move http fallback logic for `clt.go` client to a separate file.

Error handling and naming convention is now unified across recently added endpoints.

`GetNode` is added in #6539 

e PR - #255

### CODE REVIEW

**Review by andrejtokarcik — APPROVED** [review]

File: lib/auth/grpcserver.go [hunk 1]
```diff
@@ -2323,6 +2300,75 @@ func (g *GRPCServer) DeleteToken(ctx context.Context, req *types.ResourceRequest
...
+	}
+	nodes, err := auth.ServerWithRoles.GetNodes(ctx, req.Namespace)
+	if err != nil {
+		return nil, trail.ToGRPC(err)
+	}
+	serversV2 := make([]*types.ServerV2, len(nodes))
+	for i, t := range nodes {
+		var ok bool
+		if serversV2[i], ok = t.(*types.ServerV2); !ok {
+			return nil, trail.ToGRPC(trace.Errorf("encountered unexpected node type: %T", t))
```
[the remark this answers is not in the document]
Comment (Joerger) [reply lib/auth/grpcserver.go:2317]: This is an internal server error rather than a request/parameter level error.

**Review by andrejtokarcik — APPROVED** [review]

File: api/client/client.go [hunk 2]
```diff
@@ -1239,3 +1222,61 @@ func (c *Client) DeleteToken(ctx context.Context, name string) error {
 	_, err := c.grpc.DeleteToken(ctx, &types.ResourceRequest{Name: name}, c.callOpts...)
 	return trail.FromGRPC(err)
 }
+
+// GetNodes returns a list of nodes by namespace. If namespace is not specified, the default
+// namespace is used. Nodes that the user doesn't have access to are filtered out.
+func (c *Client) GetNodes(ctx context.Context, namespace string) ([]types.Server, error) {
+	if namespace == "" {
+		namespace = defaults.Namespace
+	}
```
Comment (awly) [inline api/client/client.go:1231]: DeleteNode/DeleteAllNodes treat empty namespace as an error, but here it's defaulted.
Let's make it an error everywhere.

File: lib/auth/httpfallback.go [hunk 3]
```diff
@@ -0,0 +1,640 @@
...
+		if !trace.IsNotImplemented(err) {
+			return nil, trace.Wrap(err)
+		}
+	} else {
+		return resp, nil
+	}
+
+	if namespace == "" {
+		return nil, trace.BadParameter(MissingNamespaceError)
+	}
```
Comment (awly) [inline lib/auth/httpfallback.go:609]: keep validation on the server side

**Review by awly — APPROVED** [review]

**Review by russjones — APPROVED** [review]
Bot.

**Pull request merged**